### PR TITLE
unify jar list and classpath

### PR DESCRIPTION
**Pull request by kroening**
This is part of the campaign to simplify interfaces within the Java frontend.

### CODE REVIEW

File: jbmc/src/java_bytecode/java_class_loader.h [hunk 1]
```diff
@@ -111,10 +119,21 @@ class java_class_loadert:public messaget
...
   std::string java_cp_include_files;
 
-  /// List of filesystem paths to .jar files that will be used, in the given
-  /// order, to find and load a class, provided its name (see \ref
-  /// get_parse_tree).
-  std::list<std::string> jar_files;
+  /// An entry in the classpath
+  struct classpath_entryt
+  {
+    using kindt = enum { JAR, DIRECTORY };
```
Comment (mgudemann) [inline jbmc/src/java_bytecode/java_class_loader.h:125]: `enum class` ?

File: jbmc/src/java_bytecode/java_class_loader.cpp [hunk 2]
```diff
@@ -124,55 +136,47 @@ java_class_loadert::get_parse_tree(
...
+        // Look in the given directory
+        const std::string class_file = class_name_to_file(class_name);
+        const std::string full_path =
+#ifdef _WIN32
+          cp_entry.path + '\\' + class_file;
+#else
+          cp_entry.path + '/' + class_file;
+#endif
+
+        if(!class_loader_limit.load_class_file(class_file))
```
Comment (mgudemann) [inline jbmc/src/java_bytecode/java_class_loader.cpp]: isn't this already checked above?

File: jbmc/src/java_bytecode/java_class_loader.h (hunk 1 above)
Comment (kroening) [reply jbmc/src/java_bytecode/java_class_loader.h:125]: An enum class provides no obvious benefit here -- the namespace isn't crowded.

File: jbmc/src/java_bytecode/java_class_loader.cpp (hunk 2 above)
Comment (kroening) [reply jbmc/src/java_bytecode/java_class_loader.cpp]: Indeed, removed that.

**Review by mgudemann — APPROVED** [review]

**Pull request merged**